matsim/src/main/java/org/matsim/core/scenario/ScenarioLoaderImpl.java

```diff
@@ -196,9 +200,50 @@ private void loadPopulation() {
 		if ((this.config.plans() != null) && (this.config.plans().getInputPersonAttributeFile() != null)) {
 			URL personAttributesURL = this.config.plans().getInputPersonAttributeFileURL(this.config.getContext());
 			log.info("loading person attributes from " + personAttributesURL);
-			ObjectAttributesXmlReader reader = new ObjectAttributesXmlReader(this.scenario.getPopulation().getPersonAttributes());
+//			final PersonAttributes personAttributes = this.scenario.getPopulation().getPersonAttributes();
+			final ObjectAttributes personAttributes = new ObjectAttributes() ;
+			ObjectAttributesXmlReader reader = new ObjectAttributesXmlReader( personAttributes );
 			reader.putAttributeConverters( attributeConverters );
 			reader.parse(personAttributesURL);
+
+			for( Person person : this.scenario.getPopulation().getPersons().values() ){
+				Collection<String> keys = ObjectAttributesUtils.getAllAttributeNames( personAttributes, person.getId().toString() );
+				for( String key : keys ){
+					Object value = personAttributes.getAttribute( person.getId().toString(), key );
+					person.getAttributes().putAttribute( key, value ) ;
+				}
+				personAttributes.removeAllAttributes( person.getId().toString() );
+			}
+			// (some of the above could also become a static helper method in ObjectAttributesUtils, but this here seems the only
+			// place within matsim core where the personAttributes are automatically read so maybe there is no need for this. kai, jun'19)
+
+			if ( !personAttributes.toString().equals( "" ) ) {
+				String message = "personAttributes not empty after going through all persons, meaning that it contains material for personIDs that " +
+								 "are not in the population.  This is not necessarily a bug so we will continue, but note that such material " +
+								 "will no longer be contained in the output_* files.  The material that is still there will follow.  kai, " +
+								 "jun'19" ;
+				log.warn( message ) ;
+				log.warn( personAttributes.toString() ) ;
+			}
+
+			final String outputDirectory = this.config.controler().getOutputDirectory();
+			final File outDir = new File( outputDirectory );
+			if ( outDir.exists() && outDir.canWrite() ){
```

Comment:
I do not like this. I would never expect a reader to have this kind of side-effect. In particular, this will ignore the setting that says "crash immediately if output directory exists".

For most users, taking ITERS/it.0/0.plans.xml.gz should be good enough.

Reply:
Only this: `ITERS/it.0/0.plans.xml.gz` is already routed.  So it is a totally different animal.  But I don't really care; I myself won't have any difficulties with this.